Add a new variant of xcb:unmarshal that returns an object

Stebalien opened the pull request:
Instead of having to write:

```elisp
(let* ((obj (let ((m (make-instance 'xcb:ClientMessage)))
              (xcb:unmarshal m raw-data)
              m))
       (type (slot-value obj 'type))
```

We can now write:

```elisp
(let* ((obj (xcb:unmarshal 'xcb:ClientMessage raw-data))
       (type (slot-value obj 'type))
```

We could also define a new method given that this is a bit of an abuse of generics (cc @minad). We'd just need to come up with a name for it (`xcb:make-unmarshal`, `xcb:unmarshal-instance`, `xcb:unmarshal-new`?).

Stebalien committed:
Add a new xcb:unmarshal-new that returns the object

Instead of having to write:

  (let* ((obj (let ((m (make-instance 'xcb:ClientMessage)))
                (xcb:unmarshal m raw-data)
                m))
         (type (slot-value obj 'type))

We can now write:

  (let* ((obj (xcb:unmarshal-new 'xcb:ClientMessage raw-data))
         (type (slot-value obj 'type))

minad commented:
Yes, good idea. There seems to be a lot of boilerplate code in XELB/EXWM which we could shorten with better utility functions. But I am not fond of abusing generics for overloading like this. We better introduce a new function `xcb:unmarshal-new` or `xcb:make-and-unmarshal`.

Stebalien committed:
Use xcb:unmarshal-new and simplify

* xcb-cursor.el (xcb:cursor:-parse-file): In addition to using
`xcb:unmarshal-new', replace repeated nconc with push and nreverse and
use with-slots where appropriate.
* xcb-keysyms.el (xcb:keysyms:-on-NewKeyboardNotify):
(xcb:keysyms:-on-MapNotify): Use `xcb:unmarshal-new'.
* xcb.el (xcb:-connection-setup-filter): Use `xcb:unmarshal-new', and
combine "set" statements into `setf' blocks.
(xcb:-+reply, xcb:-request-check): Use `xcb:unmarshal-new' and avoid setq.
